mise python plugin always try to compile python...

**Describe the bug**
When I try to use a python version, mise always try to compile python from sources even if the 'python_compile' setting is set to `false`



**To Reproduce**

```bash
mise use -g python@3.11

> Downloading Python-3.11.9.tar.xz...
> -> https://www.python.org/ftp/python/3.11.9/Python-3.11.9.tar.xz
> Installing Python-3.11.9...
> mise python@3.11.9 Running python-build    
 ...
```

**Expected behavior**

python should be installed from pre-compiled binaries.

**`mise doctor` output**

```text
version: 2024.4.0 linux-x64 (bcd3592 2024-04-02)
activated: yes
shims_on_path: no

build_info: 
  Target: x86_64-unknown-linux-gnu
  Features: DEFAULT, NATIVE_TLS, OPENSSL
  Built: Tue, 2 Apr 2024 01:29:57 +0000
  Rust Version: rustc 1.77.1 (7cf61ebde 2024-03-27)
  Profile: release

shell: 
  /bin/bash
  GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)
  Copyright (C) 2020 Free Software Foundation, Inc.
  License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

  This is free software; you are free to change and redistribute it.
  There is NO WARRANTY, to the extent permitted by law.

dirs: 
  data: ~/.local/share/mise
  config: ~/.config/mise
  cache: ~/.cache/mise
  state: ~/.local/state/mise
  shims: ~/.local/share/mise/shims

config_files: 
  ~/.config/mise/config.toml

plugins: 
  bun     (core)
  deno    (core)
  go      (core)
  java    (core)
  node    (core)
  python  (core)
  ruby    (core)

toolset: 
  python@3.11.9  (missing)

env_vars: 
  MISE_SHELL=bash

settings: 
  activate_aggressive = false
  all_compile = false
  always_keep_download = false
  always_keep_install = false
  asdf_compat = false
  cargo_binstall = true
  color = true
  disable_default_shorthands = false
  disable_tools = []
  experimental = false
  go_default_packages_file = "~/.default-go-packages"
  go_download_mirror = "https://dl.google.com/go"
  go_repo = "https://github.com/golang/go"
  go_set_gopath = false
  go_set_goroot = true
  go_skip_checksum = false
  jobs = 4
  legacy_version_file = true
  legacy_version_file_disable_tools = []
  node_compile = false
  not_found_auto_install = true
  paranoid = false
  plugin_autoupdate_last_check_duration = "7d"
  python_compile = false
  python_default_packages_file = "/home/lmi/.default-python-packages"
  python_pyenv_repo = "https://github.com/pyenv/pyenv.git"
  raw = false
  trusted_config_paths = []
  quiet = false
  verbose = false
  yes = false
  ci = false
  debug = false
  trace = false
  log_level = "info"
  python_venv_auto_create = false

  [status]
  missing_tools = "if_other_versions_installed"
  show_env = false
  show_tools = false

No problems found

```

**Additional context**
On ubuntu 22.04.

Hi,

I'm also experiencing this issue, have this nice docker one liner that reproduces the problem:

```
docker run -it --rm debian:12 bash -c "apt update && apt install -y git curl && curl -L --fail https://github.com/jdx/mise/releases/download/v2024.4.1/mise-v2024.4.1-linux-x64 -o mise && chmod u+x mise && ./mise use python@3.11"
```

Output truncated for brevity
```
...
Running hooks in /etc/ca-certificates/update.d...
done.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 20.9M  100 20.9M    0     0  5681k      0  0:00:03  0:00:03 --:--:-- 8120k
Downloading Python-3.11.9.tar.gz...
-> https://www.python.org/ftp/python/3.11.9/Python-3.11.9.tgz
Installing Python-3.11.9...
BUILD FAILED (Debian GNU/Linux 12 using python-build 2.4.0)
Inspect or clean up the working tree at /tmp/python-build.20240408135815.3199
Results logged to /tmp/python-build.20240408135815.3199.log
Last 10 log lines:
checking for pkg-config... no
checking for --enable-universalsdk... no
checking for --with-universal-archs... no
checking MACHDEP... "linux"
checking for gcc... no
checking for cc... no
checking for cl.exe... no
configure: error: in `/tmp/python-build.20240408135815.3199/Python-3.11.9':
configure: error: no acceptable C compiler found in $PATH
See `config.log' for more details
mise ~/.cache/mise/python/pyenv/plugins/python-build/bin/python-build failed
Downloading Python-3.11.9.tar.gz...
-> https://www.python.org/ftp/python/3.11.9/Python-3.11.9.tgz
Installing Python-3.11.9...

BUILD FAILED (Debian GNU/Linux 12 using python-build 2.4.0)

Inspect or clean up the working tree at /tmp/python-build.20240408135815.3199
Results logged to /tmp/python-build.20240408135815.3199.log

Last 10 log lines:
checking for pkg-config... no
checking for --enable-universalsdk... no
checking for --with-universal-archs... no
checking MACHDEP... "linux"
checking for gcc... no
checking for cc... no
checking for cl.exe... no
configure: error: in `/tmp/python-build.20240408135815.3199/Python-3.11.9':
configure: error: no acceptable C compiler found in $PATH
See `config.log' for more details
mise ~/.cache/mise/python/pyenv/plugins/python-build/bin/python-build exited with non-zero status: exit code 1
mise Run with --verbose or MISE_VERBOSE=1 for more information
```
As you can see, it attempts to download the tgz and build python.

Let me know if I can provide any other information

--- a/src/cli/settings/ls.rs
+++ b/src/cli/settings/ls.rs
@@ -82,7 +82,6 @@ mod tests {
         not_found_auto_install = true
         paranoid = false
         plugin_autoupdate_last_check_duration = "20m"
-        python_compile = false
         python_default_packages_file = "~/.default-python-packages"
         python_pyenv_repo = "https://github.com/pyenv/pyenv.git"
         quiet = false
@@ -125,7 +124,6 @@ mod tests {
         not_found_auto_install
         paranoid
         plugin_autoupdate_last_check_duration
-        python_compile
         python_default_packages_file
         python_pyenv_repo
         quiet

--- a/src/cli/settings/set.rs
+++ b/src/cli/settings/set.rs
@@ -147,7 +147,6 @@ pub mod tests {
         not_found_auto_install = true
         paranoid = false
         plugin_autoupdate_last_check_duration = "1"
-        python_compile = false
         python_default_packages_file = "~/.default-python-packages"
         python_pyenv_repo = "https://github.com/pyenv/pyenv.git"
         quiet = false

--- a/src/cli/settings/unset.rs
+++ b/src/cli/settings/unset.rs
@@ -64,7 +64,6 @@ mod tests {
         not_found_auto_install = true
         paranoid = false
         plugin_autoupdate_last_check_duration = "20m"
-        python_compile = false
         python_default_packages_file = "~/.default-python-packages"
         python_pyenv_repo = "https://github.com/pyenv/pyenv.git"
         quiet = false

--- a/src/config/settings.rs
+++ b/src/config/settings.rs
@@ -88,8 +88,8 @@ pub struct Settings {
     pub paranoid: bool,
     #[config(env = "MISE_PLUGIN_AUTOUPDATE_LAST_CHECK_DURATION", default = "7d")]
     pub plugin_autoupdate_last_check_duration: String,
-    #[config(env = "MISE_PYTHON_COMPILE", default = false)]
-    pub python_compile: bool,
+    #[config(env = "MISE_PYTHON_COMPILE")]
+    pub python_compile: Option<bool>,
     #[config(env = "MISE_PYTHON_DEFAULT_PACKAGES_FILE")]
     pub python_default_packages_file: Option<PathBuf>,
     #[config(env = "MISE_PYTHON_PATCH_URL")]
@@ -256,7 +256,9 @@ impl Settings {
         }
         if settings.all_compile {
             settings.node_compile = true;
-            settings.python_compile = true;
+            if settings.python_compile.is_none() {
+                settings.python_compile = Some(true);
+            }
         }
         let settings = Arc::new(settings);
         *SETTINGS.write().unwrap() = Some(settings.clone());

--- a/src/plugins/core/python.rs
+++ b/src/plugins/core/python.rs
@@ -127,6 +127,13 @@ impl PythonPlugin {
         let (tag, filename) = match precompile_info {
             Some((_, tag, filename)) => (tag, filename),
             None => {
+                if Settings::get().python_compile == Some(false) {
+                    bail!(
+                        "no precompiled python found for {}.\n\
+                        To compile python from source, run: mise settings set python_compile 1",
+                        ctx.tv.version
+                    );
+                }
                 debug!("no precompiled python found for {}", ctx.tv.version);
                 let mut available = precompiled_versions.iter().map(|(v, _, _)| v);
                 trace!("available precompiled versions: {}", available.join(", "));
@@ -305,10 +312,18 @@ impl Forge for PythonPlugin {
     }
 
     fn _list_remote_versions(&self) -> eyre::Result<Vec<String>> {
-        self.core
-            .remote_version_cache
-            .get_or_try_init(|| self.fetch_remote_versions())
-            .cloned()
+        if Settings::get().python_compile == Some(false) {
+            Ok(self
+                .fetch_precompiled_remote_versions()?
+                .iter()
+                .map(|(v, _, _)| v.clone())
+                .collect())
+        } else {
+            self.core
+                .remote_version_cache
+                .get_or_try_init(|| self.fetch_remote_versions())
+                .cloned()
+        }
     }
 
     fn legacy_filenames(&self) -> eyre::Result<Vec<String>> {
@@ -318,7 +333,7 @@ impl Forge for PythonPlugin {
     fn install_version_impl(&self, ctx: &InstallContext) -> eyre::Result<()> {
         let config = Config::get();
         let settings = Settings::try_get()?;
-        if settings.python_compile {
+        if settings.python_compile == Some(true) {
             self.install_compiled(ctx)?;
         } else {
             self.install_precompiled(ctx)?;

